### PR TITLE
Screen orientation lock: remove bad 'hidden' tests

### PR DESCRIPTION
**Pull request by marcoscaceres**
As discussed in https://github.com/w3c/screen-orientation/pull/232

### CODE REVIEW

File: screen-orientation/hidden_document.html [hunk 1]
```diff
@@ -14,48 +14,22 @@
 
   promise_test(async (t) => {
     const { minimize, restore } = window_state_context(t);
+    t.add_cleanup(restore);
```
Comment (makotokato) [inline screen-orientation/hidden_document.html:17]: `window_state_context` adds `resotre()` as clean up method. This is unnecessary. See https://github.com/web-platform-tests/wpt/pull/38031
```javascript
function window_state_context(t) {
    let rect = null;
    let state = "restored";
    t.add_cleanup(async () => {
        if (state === "minimized")
            await restore();
    });
```
I guess that you want to call `orientation.unlcok` as clean up. You should add `t.add_cleanup(makeCleanup());` instead.

File: screen-orientation/hidden_document.html [hunk 2]
```diff
@@ -14,48 +14,22 @@
...
 
     assert_equals(document.visibilityState, "hidden", "Document must be hidden");
-    await promise_rejects_dom(t, "SecurityError", screen.orientation.lock("landscape") );
+    const opposite = getOppositeOrientation();
+    await promise_rejects_dom(t, "SecurityError", screen.orientation.lock(opposite) );
   }, "hidden documents must reject went trying to call lock or unlock");
 
   promise_test(async (t) => {
     const { minimize, restore } = window_state_context(t);
+    t.add_cleanup(restore);
```
Comment (makotokato) [inline screen-orientation/hidden_document.html:28]: Please remove this since I remove this by https://github.com/web-platform-tests/wpt/pull/38031

**Review by makotokato — CHANGES_REQUESTED** [review]

File: screen-orientation/hidden_document.html (hunk 1 above)
Comment (marcoscaceres) [reply screen-orientation/hidden_document.html:17]: Ah, thanks for checking that.

Comment (marcoscaceres) [inline screen-orientation/hidden_document.html:17]:
```suggestion
    t.add_cleanup(makeCleanup());
```

File: screen-orientation/hidden_document.html [hunk 3]
```diff
@@ -14,48 +14,22 @@
...
```
Comment (marcoscaceres) [inline screen-orientation/hidden_document.html:28]:
```suggestion
```

File: screen-orientation/hidden_document.html (hunk 2 above)
Comment (marcoscaceres) [reply screen-orientation/hidden_document.html]: Oops, done!

**Review by makotokato — APPROVED** [review]

**Pull request merged**